### ISSUE TITLE
Server function wrapped in Result<'a> causes compiler warning in 5.0.0.54-preview1

### ISSUE THREAD

**Issue by allisterb** [dotnet-websharper/core#1165]
Using the 5.0.0.54 preview package from GitHub the following server function will cause a warning
````fsharp
[<Rpc>]
    let test1 (s:string) = Task.FromResult(if s.Length > 0 then Ok "foo" else  Error "0") |> Async.AwaitTask
````

4>FSC : WebSharper warning WS9002: Failed to load return type 'Microsoft.FSharp.Core.FSharpResult`2[[System.String, netstandard],[System.String, netstandard]], FSharp.Core' to verify encoding to JSON. at Mina.Web.Server.test1 `

This happens with any server function wrapped as a `Result<'a>`. If it's not wrapped then no warning occurs e.g.
````fsharp
[<Rpc>]
    let test1 (s:string) = Task.FromResult(if s.Length > 0 then "foo" else "0") |> Async.AwaitTask
````
this doesn't cause the warning.

**Comment by Jand42** [comment]
@allisterb Thanks for the notice! For the build service not to lock in assemblies, using AssemblyLoadContext was required, but I have went overboard with some changes around type load that were not necessary in the end, and was breaking loading any generic type. Fixed, will be out in version `5.0.0.56-preview1`, triggering build workflow now.

**Comment by allisterb** [comment]
Can confirm warning is gone in the latest preview.
